### PR TITLE
Feature/add java doc

### PR DESCRIPTION
**Pull request by BartoliniAndBorderCollies**
#### Issue
https://trello.com/c/PbPYlspY

### Description
This branch adds `JavaDoc` documentation to most of the application (besides integration and unit tests).
I also skipped some simple methods around different classes, because they are so simple that I believe there is no need to explain anything.

### Motivation
A well documented code is crucial for understanding any application. Furthermore me  - from the future -  or some other developer can easily get into this app if there is such a need.

### Changes introduced by this pull request
- add JavaDoc to controllers, service layer with interfaces, entities, DTOs, configuration files (including security classes), application properties files, custom exceptions handlers, repositories

@JasinskiSz  please take a look if a quality of documentaries is sufficient.

### CODE REVIEW

**Comment by JasinskiSz** [comment]
I'll respond here in general and if something requires direct reference to the code I'll respond over there.
In my opinion there is missing context in those DTOs. Documented classes are described like "This is DTO class (data transfer object) for admin" (or "admin context", sorry if I've misremembered that one). This is saying exactly nothing. DTOs are used for different purposes and in this project this is also viable thesis.
By describing a class in documentation it'd be nice if it's said why this class exists and what's its purpose. I can see that it's DTO class and I have general grasp of that from any previous project and what does DTO mean. If I'd see this project for the first time and I'd like to read documentation to learn about the project I'd learn nothing from that description.
I think I can give you an example (but it's simple one, so for your classes you can feel free to expand it for your cause):
```
/**
* A response for the REST API in {@link Controller#getMethod() Controller.getMethod}.
* It contains information about response message and status code.
* In this project its used solely to wrap those two objects and return REST API response in a standard and consistent way
* with ability to expand its functionalities beyond current implementation.
*/
public class ResponseDTO {
private String message;
private int statusCode;
// other required constructors, fields, methods
}
```

I'll let you know probably in another comments about other things if I'll find any :) right now that's all.

**Comment by BartoliniAndBorderCollies** [comment]
This is difficult to me because my assumption was that a reader knows why DTOs are used. and why they are used? because one doesn't want to include the whole entity specification of a specific class and just the parts of it serving a specific purpose.

this example also doesn't bring any important information to me:
It is just a bla bla which I wanted to avoid.

image

Anyway I will try to explain why each DTO was created.

**Comment by JasinskiSz** [comment]
As I said, reader like me know why DTOs are used, so I don't need from you this explanation. What you're missing in your description is why this exact DTO exists in the first place. I know that you don't want to send raw entity, but I don't know what purpose some DTO serves, because you didn't explain that in your javadocs.
I also disagree with your opinion that my sample doesn't bring any useful information. It says what's the reason behind creation such DTO. It says that it's used as a response (not every DTO and not in every project DTOs are used as responses in REST API only).

In any case, as I said in the beginning, this is your project and you do you here. If you disagree with my proposition of javadocs, then you can fell free to stick to your version and it's not required to have my input to every PR. I said what I believe is right here and could help you develop further in this matter. There is no reason to blindly follow my paths if you feel like your solution is better for you :)
Let me know what you'll decide and if you need further my input here as I believe there is no much time left for PRs anymore, anyway. I'll try to help as I can :)

**Comment by BartoliniAndBorderCollies** [comment]
as I try to explain why each DTO has been created and what is its purpose. I think this is important information which I keep in my mind and not here on the paper. So thank you for your review, Im going to upgrade it.
thanks!

**Comment by BartoliniAndBorderCollies** [comment]
I described DTOs better! check it out now!

File: src/main/java/com/klodnicki/Bike/v2/DTO/bike/BikeForAdminResponseDTO.java [hunk 1]
```diff
@@ -8,6 +8,13 @@
 
 import java.time.LocalDateTime;
 
+/**
+ * Data Transfer Object (DTO) for Bike information in the context of Admin operations. Since it is intended for Admin use, it holds
+ * more details than a typical DTO for a normal user. It is used as response.
+ * This class is annotated with @Data, @Builder, @AllArgsConstructor, and @NoArgsConstructor from the Lombok library to
+ * automatically generate boilerplate code like getters, setters, equals, hashcode, and toString methods,
+ * a builder pattern, and constructors with no arguments and all arguments.
+ */
```
Comment (JasinskiSz) [inline src/main/java/com/klodnicki/Bike/v2/DTO/bike/BikeForAdminResponseDTO.java:17]: Here I'd just left two first lines. I'd build further on those instead of informing that there is `@Data` or `@Builder` used as this is not something unusual. It's fine the way you did as you explained why are those here.

File: src/main/java/com/klodnicki/Bike/v2/service/api/BikeServiceApi.java [hunk 2]
```diff
@@ -8,12 +8,26 @@
 
 import java.util.List;
 
+/**
+ * This interface defines the operations for the BikeService.
+ * It extends AddOperation, FindOperation, UpdateOperation, DeleteOperation, and SaveOperation interfaces.
+ */
```
Comment (JasinskiSz) [inline src/main/java/com/klodnicki/Bike/v2/service/api/BikeServiceApi.java:14]: Here for example instead of saying what is obvious (that the class implements some interfaces), an explanation why it is important for this class to implement those and not other interfaces. I hope you understand what do I mean.

**Review by JasinskiSz — COMMENTED** [review]
We can talk further about documenting the code in better ways on our 1 on 1 session this week. What you created is not bad! But it could be better :)

And btw... Merry Christmas, @BartoliniAndBorderCollies! Hope this review finds you well!

File: src/main/java/com/klodnicki/Bike/v2/DTO/bike/BikeForAdminResponseDTO.java (hunk 1 above)
Comment (BartoliniAndBorderCollies) [reply src/main/java/com/klodnicki/Bike/v2/DTO/bike/BikeForAdminResponseDTO.java]: I see, I understand this. I will add then something more explanatory!

File: src/main/java/com/klodnicki/Bike/v2/service/api/BikeServiceApi.java (hunk 2 above)
Comment (BartoliniAndBorderCollies) [reply src/main/java/com/klodnicki/Bike/v2/service/api/BikeServiceApi.java:14]: but isn't the first line explaining this?  "This interface defines the operations for the BikeService."

Comment (BartoliniAndBorderCollies) [reply src/main/java/com/klodnicki/Bike/v2/service/api/BikeServiceApi.java:14]: I don't know, maybe something like this?

image

**Comment by BartoliniAndBorderCollies** [comment]
allrigh! I made some correction regarding Bike DTOs, please take a look! if it is OK I shall continue with other DTOs

**Review by JasinskiSz — APPROVED** [review]
